cli/cli.go

```diff
@@ -26,17 +25,30 @@ func Run() {
 			Usage:  "debug mode",
 			EnvVar: "DEBUG",
 		},
-
 		cli.StringFlag{
 			Name:  "log-level, l",
 			Value: "info",
 			Usage: fmt.Sprintf("Log level (options: debug, info, warn, error, fatal, panic)"),
 		},
+		cli.StringFlag{
+			Name:  "log-file",
+			Usage: "Logging the output to a file instead of stdout",
+		},
 	}
-
 	// logs
 	app.Before = func(c *cli.Context) error {
 		log.SetOutput(os.Stderr)
+		if c.IsSet("log-file") {
+			logDir := c.String("log-file")
+			fmt.Print(logDir)
```

Comment:
Rebased and print statement removed

On 13 January 2016 at 19:59, Victor Vieux <notifications@github.com> wrote:

> In cli/cli.go
> <https://github.com/docker/swarm/pull/1539#discussion_r49642471>:
>
> >  	// logs
> >  	app.Before = func(c *cli.Context) error {
> >  		log.SetOutput(os.Stderr)
> > +		if c.IsSet("log-file") {
> > +			logDir := c.String("log-file")
> > +			fmt.Print(logDir)
>
> please remove this print.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/swarm/pull/1539/files#r49642471>.
>